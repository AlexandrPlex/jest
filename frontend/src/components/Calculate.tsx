import * as React from 'react';

export interface IStateProps {
  value1: number;
  value2: number;
  answer: number;
}

export interface IRouterProps {
  ownProps: any;
}

export interface IDispatchProps {
  onChangeValue1(value: any): void;
  onChangeValue2(value: any): void;
  onGetAnswer(value: any): void;
}

type TProps = IStateProps & IDispatchProps;

export class Calculate extends React.Component<TProps> {
  public onHandleChangeValue1 = (event: any) => {
    return this.props.onChangeValue1(Number(event.target.value));
  }
  public onHandleChangeValue2 = (event: any) => {
    return this.props.onChangeValue2(Number(event.target.value));
  }
  public onHandleClickAnswer = () => {
    return this.props.onGetAnswer(this.props.value1 + this.props.value2);
  }
  public render() {
    return (
      <div>
        <input type='text' onChange={this.onHandleChangeValue1} /><br/>
        <input type='text' onChange={this.onHandleChangeValue2} /><br/>
        <input type='submit' onClick={this.onHandleClickAnswer} /><br/>
        {this.props.answer }
      </div>
    );
  }
}
