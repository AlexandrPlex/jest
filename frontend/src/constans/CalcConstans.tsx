export enum ICalcAction {
  CHANGE_VALUE_1 = 'CHANGE_VALUE_1',
  CHANGE_VALUE_2 = 'CHANGE_VALUE_2',
  GET_ANSWER = 'GET_ANSWER',
}
