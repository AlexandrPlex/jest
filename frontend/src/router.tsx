import * as React from 'react';
import { Route, Switch } from 'react-router-dom';

import { HomePage } from 'pages/HomePage';
import { NotFoundPage } from 'pages/NotFoundPage';

export const router = (
  <div>
    <Switch>
      <Route exact path='/' component={HomePage} />
      <Route component={NotFoundPage} />
    </Switch>
  </div>
);
