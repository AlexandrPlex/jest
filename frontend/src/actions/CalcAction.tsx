import { ICalcAction } from '../constans/CalcConstans';

interface IActionReturnState {
  type: typeof ICalcAction.CHANGE_VALUE_1 | ICalcAction.CHANGE_VALUE_2 | ICalcAction.GET_ANSWER;
  payload: number;
}

export const onChangeValue1 = (value: number): IActionReturnState => ({
  type: ICalcAction.CHANGE_VALUE_1,
  payload: value,
});

export const onChangeValue2 = (value: number): IActionReturnState => ({
  type: ICalcAction.CHANGE_VALUE_2,
  payload: value,
});

export const onGetAnswer = (value: number): IActionReturnState => ({
  type: ICalcAction.GET_ANSWER,
  payload: value,
});
