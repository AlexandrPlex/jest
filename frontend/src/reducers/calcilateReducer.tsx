import {Action} from 'redux';
import {ICalcAction} from '../constans/CalcConstans';

export interface IActionType extends Action {
  type: string;
  payload: any;
}

export interface ICalcState {
  value1: number;
  value2: number;
  answer: number;
}

export const initialState = {
  get state(): ICalcState {
    return {
      value1: 0,
      value2: 0,
      answer: 0,
    };
  },
};

export default function calcilateReducer(state: ICalcState = initialState.state, action: IActionType) {
  switch (action.type) {
    case `${ICalcAction.CHANGE_VALUE_1}`:
      return {
        ...state,
        value1: action.payload,
      };
    case `${ICalcAction.CHANGE_VALUE_2}`:
      return {
        ...state,
        value2: action.payload,
      };
    case `${ICalcAction.GET_ANSWER}`:
      return {
        ...state,
        answer: action.payload,
      };
  }
  return state;
}
