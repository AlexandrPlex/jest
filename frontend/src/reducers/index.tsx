import { combineReducers } from 'redux';

import calcilateReducer from './calcilateReducer';

export const reducers = combineReducers({
  calcilateReducer,
});
