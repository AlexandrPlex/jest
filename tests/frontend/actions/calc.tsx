import React from 'react';
import renderer from 'react-test-renderer';
import {onChangeValue1} from '../../../frontend/src/actions';
import { Calculate } from '../../../frontend/src/components/Calculate';
import {ICalcAction} from '../../../frontend/src/constans/CalcConstans';
import calcilateReducer, { initialState } from '../../../frontend/src/reducers/calcilateReducer';

describe('[Actions] todoList test', () => {

  it('reducer1', () => {
    const state = {
      ...initialState.state,
      value1: 0,
      value2: 0,
      answer: 0,
    };
    const action = {
      type: ICalcAction.CHANGE_VALUE_1,
      payload: 1,
    };
    expect(calcilateReducer(state, action)).toEqual({
      ...state,
      value1: 1,
    });
  });

  it('action1', () => {
    expect(onChangeValue1(1)).toEqual({
      type: ICalcAction.CHANGE_VALUE_1,
      payload: 1,
    });
  });

  it('snapshot1', ( ) => {
    const stateProps = {
      value1: 0,
      value2: 0,
      answer: 0,
      onChangeValue1: () => 0,
      onChangeValue2: () => 0,
      onGetAnswer: () => 0,
    };
    const tree = renderer.create( <Calculate { ...stateProps }/>);
    expect(tree).toMatchSnapshot();
  });

});
