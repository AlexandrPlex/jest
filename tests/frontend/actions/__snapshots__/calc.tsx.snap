// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Actions] todoList test snapshot1 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
  />
  <br />
  <input
    onChange={[Function]}
    type="text"
  />
  <br />
  <input
    onClick={[Function]}
    type="submit"
  />
  <br />
  0
</div>
`;
